### ISSUE TITLE
[sdk/go] Report a better error message when user passes a `nil` option to an invoke or resource 

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#4549]
We've had a few users run into issues where they pass `nil` to the variadic opts of a resource or invoke call, and it panics.  Instead, we should catch this and report a useful error message.

For example - from https://github.com/pulumi/pulumi-vsphere/issues/76 - this will panic due to the `nil` argument.

```go
computeCluster, err := vsphere.LookupComputeCluster(ctx, args, nil)
```

```
  pulumi:pulumi:Stack (okd-pulumi-vsphere-upi-upi):
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xabeb4d]
    goroutine 1 [running]:
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.(*Context).Invoke(0xc00026a200, 0x141d889, 0x31, 0x114e540, 0xc000282200, 0x114e580, 0xc000326180, 0xc00031fba8, 0x1, 0x1, ...)
        /var/home/jcallen/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.0.0/go/pulumi/context.go:155 +0x17d
    github.com/pulumi/pulumi-vsphere/sdk/v2/go/vsphere.LookupComputeCluster(0xc00026a200, 0xc000282200, 0xc00031fba8, 0x1, 0x1, 0xc00025a940, 0x0, 0x0)
        /var/home/jcallen/go/pkg/mod/github.com/pulumi/pulumi-vsphere/sdk/v2@v2.2.1/go/vsphere/getComputeCluster.go:12 +0xaf
    main.main.func1(0xc00026a200, 0x1404b92, 0x13)
        /var/home/jcallen/Development/okd-pulumi-vsphere-upi/main.go:31 +0x517
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.RunWithContext(0xc00026a200, 0x1433630, 0xc00003a1bf, 0x16)
        /var/home/jcallen/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.0.0/go/pulumi/run.go:101 +0x1ff
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.RunErr(0x1433630, 0x0, 0x0, 0x0, 0x0, 0x0)
        /var/home/jcallen/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.0.0/go/pulumi/run.go:82 +0x20a
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.Run(0x1433630, 0x0, 0x0, 0x0)
        /var/home/jcallen/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.0.0/go/pulumi/run.go:39 +0x59
    main.main()
        /var/home/jcallen/Development/okd-pulumi-vsphere-upi/main.go:12 +0x3e
    exit status 2
 
    error: an unhandled error occurred: program exited with non-zero exit code: 1
```

**Comment by k9ert** [comment]
When can we expect this to be addressed?

**Comment by stack72** [comment]
Hi all

this was fixed in #5270 - sorry it wasn't closed out at that point

Paul
